core/app/models/spree/product/scopes.rb

```diff
@@ -77,8 +77,8 @@ def self.property_conditions(property)
     #   SELECT COUNT(*) ...
     add_search_scope :in_taxon do |taxon|
       includes(:classifications).
-      where("spree_products_taxons.taxon_id" => taxon.self_and_descendants.pluck(:id)).
-      order("spree_products_taxons.position ASC")
+      where("spree_classifications.taxon_id" => taxon.self_and_descendants.pluck(:id)).
```

Comment:
Does anyone know how to get this to be arel friendly? I tried the `Spree::Classification.are_table[:taxon_id]`, however I didn't realize how some of Arels internals would hiccup when `#to_sym` was called onto it; this is probably so that Rails can lower the amount of string allocations.

Anywho, for those who want a deeper understanding of the problem, here it is:

When you call `Foo.arel_table[:bar]` you have called into the [`Arel::Table`](https://github.com/rails/rails/blob/efd2410b3973671f430f04a9ca1501d38b53acf0/activerecord/lib/active_record/core.rb#L246) and then Arel hands you back this lovely [`Arel::Attributes::Attributes`](https://github.com/rails/arel/blob/master/lib/arel/attributes/attribute.rb) (also aliased to `Arel::Attribute`). This is all nice and pretty and you go "I have an Arel object back, yippie, now lets get the SQL!" and you end up doing `Foo.arel_table[:bar].to_sql` and you get `UndefinedMethodError`. Wat?

But wait, if we stop for a second and start doing some work with the code:

```ruby
Foo.arel_table[:bar].lower.to_sql
# => "LOWER \"foo\".\"bar\""
```

OK, so somehow there is a reference to the table. Its there. It exists. All we did was call `#lower` and then it just wrapped it. It turns out that you actually need to setup the code so that there's a reference to a node. Why? Cause that's where [`#to_sql`](https://github.com/rails/arel/blob/master/lib/arel/nodes/node.rb#L46) is defined.

Its unfortunate that the docs for `#to_sql` start with `FIX ME` and its not because people want `Foo.arel_table[:bar].to_sql` but because of the AST traversal to find a relation.

If we want to be able to pump a string out, this is what the code started to look life for me after a half hour of hacking:

```ruby
collector = Arel::Collectors::SQLString.new
collector = ActiveRecord::Base.connection.visitor.accept Foor.arel_table[:bar], collector
collector.value
# => "\"foo\".\"bar\""
```

TBH, if we did a lot of this and changed up the app to use Arel in a lot more spots, we could start (badly) supporting more DBs than just MySQL, PGSQL, and SQLite. But, hey, lets just stick to doing the Rails-Way™.